src/client/tile.cpp

```diff
@@ -1175,7 +1175,18 @@ bool TextureSource::generateImagePart(std::string part_of_name,
 					core::rect<s32>(pos_from, dim),
 					video::SColor(255,255,255,255),
 					NULL);*/
-			blit_with_alpha(image, baseimg, pos_from, pos_to, dim);
+
+			v2u32 dim_dst = baseimg->getDimension();
+			if (dim != dim_dst && dim.Width == dim.Height) {
```

Comment:
Because non-square could be an invalid texture. Actually there's no reason to check for that - just resize the stuff.